shims/spark32/src/main/scala/org/apache/spark/sql/catalyst/ExtendedAnalysisException.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst
+
+import org.apache.spark.{SparkThrowable, SparkThrowableHelper}
+import org.apache.spark.annotation.Stable
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.trees.Origin
+
+/**
+ * Thrown when a query fails to analyze, usually because the query itself is invalid.
+ *
+ * @since 1.3.0
+ */
+@Stable
+class ExtendedAnalysisException protected[sql] (
```

Comment:
Can we just keep one for this new exception class? Assume they have no difference. Maybe, we can move it into a common place.

Reply:
Do you mean to move it in shim/common? This class only need in 32/33/34. And 35 is not needed. It maybe not make sense to move to shim/common.

Reply:
Not a strong feeling on this. Keeping the current change is OK to me. BTW, it would be better to also add some comment for these classes to let people know why they are introduced. Thanks!